fireloop/index.d.ts

```diff
@@ -0,0 +1,595 @@
+     * @property {Array} inArray Property must match one of the values in the array to be valid.
+     * @property {String} message Optional error message if property is not valid.   Default error message: "is not included in the list".
+     */
+    validatesInclusionOf(propertyName: String): void;
+
+    /**
+     * Validate exclusion.  Require a property value not be in the specified array
+     *
+     * @param {String} propertyName  Property name to validate.
+     * @options {Object} Options
+     * @property {Array} inArray Property must match one of the values in the array to be valid.
+     * @property {String} message Optional error message if property is not valid.  Default error message: "is reserved".
+     */
+    validatesExclusionOf(propertyName: String): void;
+
+    /**
+     * Validate format. Require a model to include a property that matches the given format.
+     *
+     * Require a model to include a property that matches the given format.  Example:
+     * `User.validatesFormat('name', {with: /\w+/});`
+     *
+     * @param {String} propertyName  Property name to validate.
+     * @options {Object} Options
+     * @property {RegExp} with Regular expression to validate format.
+     * @property {String} message Optional error message if property is not valid.  Default error message: " is invalid".
+     */
+    validatesFormatOf(propertyName: String): void;
+
+    /**
+     * Validate using custom validation function.
+     *
+     * @param {String} propertyName  Property name to validate.
+     * @param {Function} validatorFn Custom validation function.
+     * @options {Object} Options See below.
+     * @property {String} message Optional error message if property is not valid.  Default error message: " is invalid".
+     */
+    validate(propertyName: string, validatorFn: Function): void;
+
+    /**
+     * Validate using custom asynchronous validation function.
+     *
+    * @param {String} propertyName  Property name to validate.
+    * @param {Function} validatorFn Custom validation function.
+    * @options {Object} Options See below
+    * @property {String} message Optional error message if property is not valid.  Default error message: " is invalid".
+    */
+    validateAsync(propertyName: String, validatorFn: Function): void;
+
+    /**
+     * Validate uniqueness. Ensure the value for property is unique in the collection of models.
+     * Not available for all connectors. Currently supported with these connectors:
+     *  - In Memory
+     *  - Oracle
+     *  - MongoDB
+     *
+     * ```
+     * // The login must be unique across all User instances.
+     * User.validatesUniquenessOf('login');
+     *
+     * // Assuming SiteUser.belongsTo(Site)
+     * // The login must be unique within each Site.
+     * SiteUser.validateUniquenessOf('login', { scopedTo: ['siteId'] });
+     * ```
+    
+    * @param {String} propertyName  Property name to validate.
+    * @options {Object} Options See below.
```

Comment:
If this method takes an options object, express that in the signature, not in jsdoc. Don't need type declarations in the jsdoc if it's in the code.